pkg/kubelet/kuberuntime/kuberuntime_container_linux.go

```diff
@@ -168,6 +168,13 @@ func (m *kubeGenericRuntimeManager) configureContainerSwapResources(lcr *runtime
 		return
 	}
 	swapConfigurationHelper := newSwapConfigurationHelper(*m.machineInfo)
+	if m.memorySwapBehavior == kubelettypes.LimitedSwap {
+		if !isCgroup2UnifiedMode() {
+			klog.InfoS("LimitedSwap is only available with cgroupsv2", "swapBehavior", m.memorySwapBehavior, "pod", klog.KObj(pod), "containerName", container.Name)
+			swapConfigurationHelper.ConfigureNoSwap(lcr)
+			return
+		}
+	}
 
 	if !utilfeature.DefaultFeatureGate.Enabled(kubefeatures.NodeSwap) {
```

Comment:
move the feature gate check to the top of the function, otherwise we'll be outputting lots of lots per container unnecessarily

Reply:
I removed logs here.